tools/topology/sof-hda-generic-kwd.m4

```diff
@@ -0,0 +1,192 @@
+# Topology for SKL+ HDA Generic machine
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include bxt DSP configuration
+include(`platform/intel/bxt.m4')
+
+# Define pipeline id for intel-generic-dmic.m4
+# to generate dmic setting
+
+ifelse(CHANNELS, `0', ,
+`
+define(KFBM_TYPE, `kfbm')
+define(DMIC_PCM_48k_ID, `6')
+define(DMIC_PCM_16k_ID, `7')
+define(DMIC_PIPELINE_48k_ID, `10')
+define(DMIC_PIPELINE_16k_ID, `11')
+define(DMIC_PIPELINE_KWD_ID, `12')
+define(DMIC_DAI_LINK_48k_ID, `6')
+define(DMIC_DAI_LINK_16k_ID, `7')
+define(KWD_PIPE_SCH_DEADLINE_US, 20000)
```

Comment:
Where are the these macros used ?

Reply:
@lgirdwood No, this is some define for the template `include(`platform/intel/intel-generic-dmic-kwd.m4')`
This will save tons of time of adding DMIC with KWD, you only need to define some pipeline/pcm/dai id.

Reply:
Ok, can you comment this. I would also add some error detection in generic-dmic-kwd.m4 if any of these mandatory macros are undefined.

Reply:
@lgirdwood Updated